src/SFA.DAS.Authorization.UnitTests/AuthorizationContextCacheTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Testing;
+
+namespace SFA.DAS.Authorization.UnitTests
+{
+    [TestFixture]
+    [Parallelizable]
+    public class AuthorizationContextCacheTests : FluentTest<AuthorizationContextCacheTestsFixture>
+    {
+        [Test]
+        public void GetAuthorizationContext_WhenGettingAuthorizationContext_ThenShouldReturnAuthorizationContext()
+        {
+            Test(f => f.GetAuthorizationContext(), (f, r) => r.Should().NotBeNull());
+        }
+
+        [Test]
+        public void GetAuthorizationContext_WhenGettingAuthorizationContext_ThenShouldGetAuthorizationContextFromAuthorizationContextProvider()
+        {
+            Test(f => f.GetAuthorizationContext(), f => f.AuthorizationContextProvider.Verify(p => p.GetAuthorizationContext(), Times.Once));
+        }
+
+        [Test]
+        public void GetAuthorizationContext_WhenGettingAuthorizationContextMultipleTimes_ThenShouldGetAuthorizationContextFromAuthorizationContextProviderOnce()
+        {
+            Test(f => f.GetAuthorizationContext(), f => f.AuthorizationContextProvider.Verify(p => p.GetAuthorizationContext(), Times.Once));
+        }
+
+        [Test]
+        public void GetAuthorizationContext_WhenGettingAuthorizationContextMultipleTimes_ThenShouldReturnSameAuthorizationContext()
+        {
+            Test(f => f.GetAuthorizationContext(3), (f, r) => f.AuthorizationContexts.ForEach(c => c.Should().Be(r)));
+        }
+    }
+
+    public class AuthorizationContextCacheTestsFixture
+    {
+        public Mock<IAuthorizationContextProvider> AuthorizationContextProvider { get; set; }
+        public IAuthorizationContextProvider AuthorizationContextCache { get; set; }
+        public List<IAuthorizationContext> AuthorizationContexts { get; set; }
+
+        public AuthorizationContextCacheTestsFixture()
+        {
+            AuthorizationContextProvider = new Mock<IAuthorizationContextProvider>();
+            AuthorizationContextCache = new AuthorizationContextCache(AuthorizationContextProvider.Object);
+            AuthorizationContexts = new List<IAuthorizationContext>();
+
+            AuthorizationContextProvider.Setup(p => p.GetAuthorizationContext()).Returns(() => new AuthorizationContext());
+        }
+
+        public IAuthorizationContext GetAuthorizationContext(int? count = 1)
```

Comment:
This method could return a `List<IAuthorizationContext>`, then `GetAuthorizationContext_WhenGettingAuthorizationContextMultipleTimes_ThenShouldReturnSameAuthorizationContext()` could directly check the result, the property `AuthorizationContexts` could be removed from the fixture (and the first test would have to check 1 non null was returned).

Not a lot in it really, just thinking it'd be nicer to check the result directly (and only assert using fixture properties for call verifications).

Reply:
Nice, fixed.